cave/com.raytheon.viz.warnings/src/com/raytheon/viz/warnings/rsc/AbstractWWAResource.java

```diff
@@ -420,47 +449,59 @@ protected void paintInternal(IGraphicsTarget target,
             }
         }
         int index = info.getFrameIndex();
-        if (!this.recordsToLoad.isEmpty()) {
-            this.updateDisplay(target);
-        }
+        boolean framesChanged = false;
+        //only do the frame logic if the frame has changed
+        if(currentFrameIdx == Integer.MIN_VALUE || currentFrameIdx != index) {
+            framesChanged = true;
+            currentFrameIdx = index;
+            currentCandidates.clear();
+
+            if (!this.recordsToLoad.isEmpty()) {
+                this.updateDisplay(target);
+            }
 
-        DataTime thisFrameTime = null;
-        if (index > -1 && index < frames.length) {
-            thisFrameTime = frames[index];
-        }
-        if (thisFrameTime == null) {
-            return;
-        }
```

Comment:
Does this section need to be in here? It looks like just formatting changes

Reply:
GitHub does a really poor job of the diff. A few sections are put into new conditionals (if statements) so that's why it looks like just a white space change. If you look closer you should be able to see the change.
Let me know if you want me to walk you through it?